WinUI3 Settings page logs

herve-er opened the pull request:
Add Log related settings
**depends on #1171** 
https://github.com/user-attachments/assets/e8e82e5a-bd3e-40cc-9c07-68a23f286262

herve-er committed:
feat(logging Settings WinUI3): Add debug log management UI

- Added `LogFilePath` and `LogFolder` properties in `Logger.cs`.
- Updated `SettingsPage.xaml` with new controls for log management:
  - Added `LogSettingsExpander` with options for enabling/disabling logs.
  - Replaced old controls with `ToggleSwitch` and `ComboBox`.
  - Added a card to open the debug folder with a new icon.
- Implemented event handlers in `SettingsPage.xaml.cs` for new controls:
  - `OpenDebugFolderButton_Click`, `EnableDebugLogsToggleSwitch_Toggled`,
    `PurgeOldLogsToggleSwitch_Toggled`, `ExtendedLogToggleSwitch_Toggled`,
    and `LogLevelComboBox_SelectionChanged`.
- Added `users-circle-question-mark` icon in `InfomaniakStylesStatic.xaml`.
- Updated `Settings.cs` ViewModel:
  - Added `LogEnabled` and `ExtendedLogEnabled` properties.
  - Updated `LogLevel` to synchronize with new properties.
  - Added `ChangeLogLevel` and `ChangePurgeOldLog` methods.
- Added `users-circle-question-mark.svg` for debug log representation.
- Improved UI/UX for log management and added error handling.

herve-er committed:
feat(settings): Added localization for logs settings

Added unique x:Uid identifiers to UI elements in `SettingsPage.xaml`
to support localization. Replaced static texts with resource keys
from `Resources.resw`. Enhanced ComboBox localization for debug
levels. Introduced new localized strings for headers, descriptions,
and tooltips.

ClementKunz commented on src/gui4/windows/kDrive client/kDrive client/Pages/SettingsPage.xaml:
```diff
@@ -402,40 +402,70 @@
+                                <ComboBox SelectedValue="{x:Bind ViewModel.Settings.LogLevel, Converter={StaticResource ToStringConverter}, Mode=OneWay}"
+                                          SelectedValuePath="Tag"
+                                          SelectionChanged="LogLevelComboBox_SelectionChanged">
+                                    <ComboBoxItem Tag="Extended"
```
Do we need the `Extended` level here since we already have a dedicated `ToggleSwitch`? Should we keep only 1?

herve-er replied:
Done here: [feat(settings): remove "None" debug level option](https://github.com/Infomaniak/desktop-kDrive/pull/1174/commits/0fe43005ab7fdaf1fea54dc36237cbb7ed8398b8)

ClementKunz commented on src/gui4/windows/kDrive client/kDrive client/Pages/SettingsPage.xaml:
```diff
@@ -402,40 +402,70 @@
+                                                  x:Uid="Page_SettingsPage_Logs_DebugLevel_ComboBox_Error" />
+                                    <ComboBoxItem Tag="Fatal"
+                                                  x:Uid="Page_SettingsPage_Logs_DebugLevel_ComboBox_Fatal" />
+                                    <ComboBoxItem Tag="None"
```
Same idea: selecting `None` is the same as disabling the logs completely

ClementKunz commented on src/gui4/windows/kDrive client/kDrive client/Pages/SettingsPage.xaml.cs:
```diff
@@ -308,6 +307,77 @@ private void CancelProxySettingsButton_Click(object sender, RoutedEventArgs e)
+
+            bool toggleIsOn = toggleSwitch?.IsOn ?? false;
+
+            if (ViewModel.Settings.LogLevel == Logger.Level.None && !toggleIsOn) return; // No change needed
+            if(ViewModel.Settings.LogLevel != Logger.Level.None && toggleIsOn) return; // No change needed
```
Is there a way to format automatically on VS? That would be nice :)

herve-er committed:
feat(settings): remove "None" debug level option

Removed the "None" option from the debug level dropdown in `SettingsPage.xaml`.
Deleted associated resources in `Resources.resw`, including the key
`Page_SettingsPage_Logs_DebugLevel_ComboBox_None.Content`. Updated remaining
resources to ensure consistency in the debug level options.
